src/mock_parsec/mod.rs

```diff
@@ -0,0 +1,295 @@
+        &self.our_id.public_id()
+    }
+
+    pub fn vote_for(&mut self, observation: Observation<T, S::PublicId>) -> Result<(), Error> {
+        let holder = ObservationHolder::new(
+            observation.clone(),
+            self.our_id.public_id(),
+            self.consensus_mode,
+        );
+
+        state::with(self.section_hash, |state| {
+            if let Some(block) = state
+                .observations
+                .entry(holder.clone())
+                .or_insert_with(ObservationState::new)
+                .vote(
+                    &self.our_id,
+                    &self.peer_list,
+                    self.consensus_mode,
+                    observation,
+                )
+            {
+                state.blocks.push((block, holder.clone()))
+            }
+        });
+
+        self.observations
+            .entry(holder)
+            .or_insert_with(ObservationInfo::new)
+            .our = true;
+        self.update_blocks();
+
+        Ok(())
+    }
+
+    pub fn gossip_recipients(&self) -> impl Iterator<Item = &S::PublicId> {
+        let iter = if self.peer_list.contains(self.our_id.public_id()) {
+            Some(
+                self.peer_list
+                    .iter()
+                    .filter(move |peer_id| *peer_id != self.our_id.public_id()),
+            )
+        } else {
+            None
+        };
+
+        iter.into_iter().flatten()
+    }
+
+    pub fn create_gossip(
+        &self,
+        _peer_id: Option<&S::PublicId>,
+    ) -> Result<Request<T, S::PublicId>, Error> {
+        Ok(Request::new())
+    }
+
+    pub fn handle_request(
+        &mut self,
+        _src: &S::PublicId,
+        _req: Request<T, S::PublicId>,
+    ) -> Result<Response<T, S::PublicId>, Error> {
+        self.update_blocks();
+        Ok(Response::new())
+    }
+
+    pub fn handle_response(
+        &mut self,
+        _src: &S::PublicId,
+        _resp: Response<T, S::PublicId>,
+    ) -> Result<(), Error> {
+        self.update_blocks();
+        Ok(())
+    }
+
+    pub fn poll(&mut self) -> Option<Block<T, S::PublicId>> {
+        state::with(self.section_hash, |state| {
+            if let Some((block, holder)) = state.blocks.get(self.first_unpolled) {
+                self.first_unpolled += 1;
+                self.observations
+                    .entry(holder.clone())
+                    .or_insert_with(ObservationInfo::new)
+                    .state = ConsensusState::Polled;
+
+                Some(block.clone())
+            } else {
+                None
+            }
+        })
+    }
+
+    #[allow(unused)]
+    pub fn can_vote(&self) -> bool {
+        unimplemented!()
+    }
+
+    #[allow(unused)]
+    pub fn have_voted_for(&self, _observation: &Observation<T, S::PublicId>) -> bool {
+        unimplemented!()
+    }
+
+    pub fn has_unconsensused_observations(&self) -> bool {
+        state::with::<T, S::PublicId, _, _>(self.section_hash, |section_state| {
```

Comment:
I'd rather call this one `state`, and the one below `observation_state`, since the section state is called `state` in the other methods, too.